pagopa/io-app
# feat(Sicilia Vola): [IASV-12] SiciliaVola api integration

## Short description
This PR adds the network logic and saga to handle both the SiciliaVola flows: 
- Voucher generation 
- Voucher list
The types corresponding to networking actions and their reducers have also been updated where needed in this PR.

Note that this PR does not complete all that is needed for networking. In fact, due to the wrong specifications some APIs have been managed as mock.
Furthermore, the local swagger has been modified manually by adding the authentication header to some calls and creating parent types for the arrays (the latter due to a problem on our decoder generation library)

## List of changes proposed in this pull request
- Added client
- Added saga handler
- Updated types in the asyncAction where is needed and the correspondents reducers

## How to test
Make sure you have the dev server master branch up to date. 
Use the custom command to dispatch the following actions:
- `svServiceAlive.request` 
- `svTosAccepted.request`
- `svAcceptTos.request`
- `svGenerateVoucherAvailableState.request`
- `svGenerateVoucherAvailableRegion.request`
- `svGenerateVoucherAvailableProvince.request`
- `svGenerateVoucherAvailableMunicipality.request`
- `svGenerateVoucherAvailableDestination.request`
- `svGenerateVoucherAvailableDestinationAbroad.request`
- `svGenerateVoucherGeneratedVoucher.request`
- `svVoucherListGet.request`
- `svVoucherDetailGet.request`
- `svVoucherRevocation.request`
